File: develnext-designer/src/main/resources/JPHP-INF/sdk/php/gui/dock/UXDockNode.php
<?php
namespace php\gui\dock;

use php\gui\layout\UXVBox;
use php\gui\UXNode;

/**
 * Class UXDockPane
 * @package php\gui\dock
 */
class UXDockNode extends UXVBox
{
    /**
     * @var string
     */
    public $title;

    /**
     * @var string
     */
    public $graphic;

    /**
     * @var UXNode
     */
    public $content;

    /**
     * @var bool
     */
    public $maximized;

    /**
     * @var bool
     */
    public $minimized;

    /**
     * @var bool
     */
    public $resizable;

    /**
     * @var bool
     */
    public $closable;

    /**
     * @var bool
     */
    public $floatable;

    /**
     * @var bool
     */
    public $minimizable;

    /**
     * UXDockNode constructor.
     * @param UXNode $content
     * @param string $title
     * @param UXNode|null $graphic
     */
    public function __construct(UXNode $content, $title = '', UXNode $graphic = null)
    {
    }

    /**
     * @return bool
     */
    public function isClosed()
    {
    }

    /**
     * @return bool
     */
    public function isTabbed()
    {
    }

    /**
     * @return bool
     */
    public function isDocked()
    {
    }

    /**
     * @return bool
     */
    public function isMouseResizeZone()
    {
    }

    /**
     * @param UXDockPane $pane
     * @param string $dockPos
     * @param UXNode $sibling (optional)
     */
    public function dock(UXDockPane $pane, $dockPos, UXNode $sibling)
    {
    }

    /**
     * ...
     */
    public function undock()
    {
    }

    public function close()
    {
    }

    public function focus()
    {
    }
}

File: develnext-designer/src/main/java/org/develnext/jphp/gui/designer/classes/dock/UXDockNode.java
/*package org.develnext.jphp.gui.designer.classes.dock;

import javafx.geometry.Pos;
import javafx.scene.Node;
import javafx.scene.layout.VBox;
import org.develnext.jphp.ext.javafx.classes.layout.UXStackPane;
import org.develnext.jphp.ext.javafx.classes.layout.UXVBox;
import org.develnext.jphp.gui.designer.GuiDesignerExtension;
import org.dockfx.DockNode;
import org.dockfx.DockPane;
import org.dockfx.DockPos;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(GuiDesignerExtension.NS_DOCK)
public class UXDockNode extends UXVBox<DockNode> {
    interface WrappedInterface {
        @Property String title();
        @Property Node graphic();

        @Property boolean closable();
        @Property boolean floatable();
        @Property boolean floating();
        @Property boolean minimizable();
        @Property("resizable") boolean stageResizable();

        @Property boolean maximized();
        @Property boolean minimized();

        @Property("content") Node contents();

        void close();
        void focus();

        void dock(DockPane dockPane, DockPos dockPos);
        void dock(DockPane dockPane, DockPos dockPos, Node sibling);
        void undock();

        Node getLastDockSibling();
        DockPos getLastDockPos();

        boolean isClosed();
        boolean isTabbed();
        boolean isMouseResizeZone();
        boolean isDocked();
    }

    public UXDockNode(Environment env, DockNode wrappedObject) {
        super(env, wrappedObject);
    }

    public UXDockNode(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct(Node content) {
        __wrappedObject = new DockNode(content);
    }

    @Signature
    public void __construct(Node content, String title) {
        __wrappedObject = new DockNode(content, title);
    }

    @Override
    public DockNode getWrappedObject() {
        return (DockNode) super.getWrappedObject();
    }

    @Signature
    public void __construct(Node content, String title, Node graphics) {
        __wrappedObject = new DockNode(content, title, graphics);
    }
}
*/